SDK_versioned_docs/version-3.0.0/guides/auto-router/swap-and-add.md

```diff
@@ -0,0 +1,147 @@
+---
+id: swap-and-add
+title: Swap and Add Liquidity Atomically
+sidebar_position: 2
+---
+
+# Swap and Add Liquidity Atomically
+
+You can use the Auto Router to fetch optimized trade routes that result in token balances proportional to the amount needed for adding optimal liquidity to a pool position. Use the method `routeToRatio` to get quotes, gas information, and calldata for atomically swapping tokens to a particular ratio and adding the new token balances to a pool position.
+
+## Initializing the Alpha Router
+
+Follow the instructions to [import](https://docs.uniswap.org/sdk/guides/auto-router/quick-start#importing-the-package) then [initialize](https://docs.uniswap.org/sdk/guides/auto-router/quick-start#initializing-the-alpharouter) an instance of the Alpha Router
+
+## Calling `routeToRatio`
+
+The `routeToRatio` method returns all the swap and add calldata needed for submitting an atomic transaction to perform a swap and add liquidity to a position.
+
+Once you instantiate `AlphaRouter` call `routeToRatio` with the following parameters:
+
+```typescript
+{
+	token0Balance: CurrencyAmount,
+	token1Balance: CurrencyAmount,
+	position: Position,
+	swapAndAddConfig: SwapAndAddConfig,
+	swapAndAddOptions?: swapAndAddOptions,
+	partialRoutingConfig?: Partial<AlphaRouterConfig> = {}
+}
+```
+
+#### Parameters
+
+`token0Balance` [required]
+
+- The initial starting balance of token0 of the pool for which to add liquidity
+
+`token1Balance` [required]
+
+- The initial starting balance of token1 of the pool for which to add liquidity
+
+`position` [required]
+
+- A position object that contains the details of the position for which to add liquidity. The position liquidity can be set to 1, since liquidity is still unknown and will be set inside the call to `routeToRatio`
+
+`swapAndAddConfig` [required]
+
+- Configurations for the routeToRatio algorithm. `ratioErrorTolerance` determines the margin of error the resulting ratio can have from the optimal ratio. `maxIterations` determines the maximum times the algorithm will iterate to find a ratio within error tolerance. If max iterations is exceeded, an error is returned.
+
+`swapAndAddOptions` [optional]
+
+- If included, routeToRatio will return the calldata for executing the atomic swap-and-add. These options contain `swapConfig` and `addLiquidityOptions`. `swapConfig` configures to set a recipient of leftover dust from swap, slippageTolerance, deadline, and inputTokenPermit. `addLiquidityOptions` contains slippageTolerance and deadline for the swap. `addLiquidityOptions` must contain a `tokenId` to add to an existing position, or `recipient` to mint a new one. It also includes a slippage tolerance and deadline for adding liquidity.
```

Comment:
typo? in the code example, swapConfig and addLiquidityOptions sit under `swapAndAddConfig`  not `swapAndAddOptions`